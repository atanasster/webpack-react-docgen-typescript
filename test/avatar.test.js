import compiler from './compiler.js';

jest.setTimeout(100000);

test('avatar', async () => {
  const stats = await compiler('./examples/Avatar.tsx');
  const jsonStats = stats.toJson();
  const output = jsonStats.modules[2].source;
  expect(output).toMatchSnapshot();
});
