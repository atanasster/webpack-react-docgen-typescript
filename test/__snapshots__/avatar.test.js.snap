// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar 1`] = `
"import _objectWithoutProperties from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/objectWithoutProperties\\";
var _jsxFileName = \\"/Users/atanasster/webpack-react-docgen-typescript/test/examples/Avatar.tsx\\";
import React from 'react';

/**
* An image with some text, can be used as a user Avatar image<br/>
* \`import { Avatar } from 'grommet-controls;'\`<br/>
* \`<Avatar image='...' />\`<br/>
*/
export var Avatar = function Avatar(_ref) {
  var image = _ref.image,
      title = _ref.title,
      subTitle = _ref.subTitle,
      rest = _objectWithoutProperties(_ref, [\\"image\\", \\"title\\", \\"subTitle\\"]);

  return React.createElement(\\"div\\", {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12
    },
    __self: this
  }, React.createElement(\\"img\\", Object.assign({
    src: image
  }, rest, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13
    },
    __self: this
  })), React.createElement(\\"div\\", {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17
    },
    __self: this
  }, React.createElement(\\"label\\", {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18
    },
    __self: this
  }, title), React.createElement(\\"label\\", {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19
    },
    __self: this
  }, subTitle)));
};

try {
  Avatar.__docgenInfo = {
    \\"description\\": \\"An image with some text, can be used as a user Avatar image<br/>\\\\n\`import { Avatar } from 'grommet-controls;'\`<br/>\\\\n\`<Avatar image='...' />\`<br/>\\",
    \\"displayName\\": \\"Avatar\\",
    \\"methods\\": [],
    \\"props\\": {
      \\"image\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"Image to be displayed\\",
        \\"name\\": \\"image\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/AvatarProps.ts\\",
          \\"name\\": \\"IAvatarProps\\"
        },
        \\"required\\": false,
        \\"type\\": {
          \\"name\\": \\"string\\"
        }
      },
      \\"title\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"Main title(name) for the Avatar\\",
        \\"name\\": \\"title\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/AvatarProps.ts\\",
          \\"name\\": \\"IAvatarProps\\"
        },
        \\"required\\": false,
        \\"type\\": {
          \\"name\\": \\"string\\"
        }
      },
      \\"subTitle\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"Secondary title for the Avatar\\",
        \\"name\\": \\"subTitle\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/AvatarProps.ts\\",
          \\"name\\": \\"IAvatarProps\\"
        },
        \\"required\\": false,
        \\"type\\": {
          \\"name\\": \\"string\\"
        }
      }
    },
    \\"type\\": \\"tsType\\"
  };
} catch (e) {//eat exception
}"
`;
